[enhancement] Nav-level increase when cloning sub-menus

petrklus opened the pull request:
Please find proposed amend to ResponsiveNav.js to better support multi-level responsive menus. When data-o-hierarchical-nav-is-cloneable is specified for a parent, the script ensures that the nav-level of the cloned submenu is increased to compensate for it being placed under the "More/Menu" elements.

petrklus commented:
Demo: http://cnkpreview.s3.amazonaws.com/ft1_origami/o-hierarchical-nav/responsive-nav-cloneable.html

petrklus commented:
However, while the above works - I would imagine there may be better ways of tagging the menus to either be preserved/flattened.

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -64,6 +64,10 @@ function ResponsiveNav(rootEl) {
 		prefixIds(el);
+        
+        // increase level of menus
+        incrementMenuDepths(cloneEl)
```
indentation and semicolon missing

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
 	}
 
+	function incrementMenuDepths(el) {
+		// id's are prefixed to ensure that any id based functionality uses the visible element
```
what ids?

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
+				if (child instanceof HTMLElement) {
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {  
+                        // increment level
+                        let oldLevel = isNaN(parseInt(child.getAttribute('data-o-hierarchical-nav-level'))) ? 0 : parseInt(child.getAttribute('data-o-hierarchical-nav-level'))              
```
Indentation and might be worth taking this out to a separate variable: `parseInt(child.getAttribute('data-o-hierarchical-nav-level'))`

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {  
+                        // increment level
+                        let oldLevel = isNaN(parseInt(child.getAttribute('data-o-hierarchical-nav-level'))) ? 0 : parseInt(child.getAttribute('data-o-hierarchical-nav-level'))              
+                        child.setAttribute('data-o-hierarchical-nav-level', 1+oldLevel);
```
to follow the general coding style, could this be `oldLevel + 1`?

AlbertoElias commented:
Thanks a lot @petrklus I left a few comments and I've got another question, do the nav levels go back to normal?

petrklus commented:
Thank you very much for your comments @AlbertoElias! I have fixed the whitespace issues (I guess Atom's auto-detection is not as smart as it could be) and also updated the comments and added a variable for the nav depth. Regarding your other question - there should not be a need for the nav levels to go back, as we are only operating on cloned elements, not the originals.

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -64,10 +64,10 @@ function ResponsiveNav(rootEl) {
-        
+
+    // increase level of nested menus
+    incrementMenuDepths(cloneEl);
```
still doesn't seem properly indented

AlbertoElias commented on src/js/ResponsiveNav.js:
```diff
@@ -80,19 +80,20 @@ function ResponsiveNav(rootEl) {
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {
+						// increment nav-level when attribute present
+						let origNavLevel = parseInt(child.getAttribute('data-o-hierarchical-nav-level'))
+            let updatedNavLevel = (isNaN(origNavLevel) ? 0 : origNavLevel) + 1
```
indentation

AlbertoElias commented:
Fair point, there´s still an indentation issue and you've also got merge conflicts. Thanks!

petrklus commented:
Just had a look at the file and the indentation looks fine to me (https://github.com/cnk-digital-solutions/o-hierarchical-nav/blob/master/src/js/ResponsiveNav.js) - could you please confirm?

AlbertoElias commented:
True! Just the merge conflicts then :)

petrklus commented:
@AlbertoElias should be all there!
